pkg/prometheus/server/statefulset.go

```diff
@@ -302,6 +302,12 @@ func makeStatefulSetSpec(
 		promArgs = append(promArgs, monitoringv1.Argument{Name: "storage.tsdb.min-block-duration", Value: thanosBlockDuration})
 	}
 
+	if cpf.TSDB != nil && cpf.TSDB.OutOfOrderTimeWindow != "" &&
```

Comment:
Can we add a comment explaining the rationale here? A link to the original issue would also work.

Reply:
Thanks @simonpasquier .
Add ref issue link done.